python/cudf/cudf/core/window/rolling.py

```diff
@@ -4,6 +4,7 @@
 
 import numba
 import pandas as pd
+from pandas.api.indexers import BaseIndexer
```

Comment:
Should we make our own `BaseIndexer` class whose `get_window_bounds()` method is expected to return a tuple of CuPy arrays instead?

Reply:
we can, but we would also have to support pandas `BaseIndexer` implementations too as `dask-sql` is not aware of it.

Reply:
Hmm -- yeah I was wondering if dask-sql could use the right `BaseIndexer` depending on whether it's using CPU/GPU, but that's definitely out of scope for this discussion.

For here, I'd say yeah -- let's at provide a `cudf.BaseIndexer` and support both.

Reply:
Had a discussion offline and opened a separate feature request to track it: https://github.com/rapidsai/cudf/issues/9105